tests/FunctionalTestCase.php

```diff
@@ -53,23 +54,32 @@ protected function getFeatureCompatibilityVersion(ReadPreference $readPreference
             return $this->getServerVersion($readPreference);
         }
 
-        $cursor = $this->manager->executeCommand(
-            'admin',
-            new Command(['getParameter' => 1, 'featureCompatibilityVersion' => 1]),
-            $readPreference ?: new ReadPreference(ReadPreference::RP_PRIMARY)
-        );
-
-        $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
-        $document = current($cursor->toArray());
-
-        // MongoDB 3.6: featureCompatibilityVersion is an embedded document
-        if (isset($document['featureCompatibilityVersion']['version']) && is_string($document['featureCompatibilityVersion']['version'])) {
-            return $document['featureCompatibilityVersion']['version'];
-        }
-
-        // MongoDB 3.4: featureCompatibilityVersion is a string
-        if (isset($document['featureCompatibilityVersion']) && is_string($document['featureCompatibilityVersion'])) {
-            return $document['featureCompatibilityVersion'];
+        try {
+            $cursor = $this->manager->executeCommand(
+                'admin',
+                new Command(['getParameter' => 1, 'featureCompatibilityVersion' => 1]),
+                $readPreference ?: new ReadPreference(ReadPreference::RP_PRIMARY)
+            );
+
+            $cursor->setTypeMap(['root' => 'array', 'document' => 'array']);
+            $document = current($cursor->toArray());
+
+            // MongoDB 3.6: featureCompatibilityVersion is an embedded document
+            if (isset($document['featureCompatibilityVersion']['version']) && is_string($document['featureCompatibilityVersion']['version'])) {
+                return $document['featureCompatibilityVersion']['version'];
+            }
+
+            // MongoDB 3.4: featureCompatibilityVersion is a string
+            if (isset($document['featureCompatibilityVersion']) && is_string($document['featureCompatibilityVersion'])) {
+                return $document['featureCompatibilityVersion'];
+            }
+        } catch (CommandException $e) {
+            /* It is not possible to get the featureCompatibilityVersion from mongos, so if the
+             * command fails, fall back to the server version if the exception's code signals it was
+             * a "no option found to get" error. */
+            if ($e->getCode() == 17) {
+                return $this->getServerVersion($readPreference);
+            }
```

Comment:
As-is, I think this `catch` should re-throw the exception if we don't return the server version. Otherwise, there is no return value and things might fail unexpectedly. Rethrowing would lead to a much clearer test failure. That said, I think my suggestion above (i.e. use an early conditional) may make this a moot point.